core/src/rpc_subscriptions.rs

```diff
@@ -313,11 +368,111 @@ impl RpcSubscriptions {
     }
 
     pub fn notify_slot(&self, slot: Slot, parent: Slot, root: Slot) {
-        let subscriptions = self.slot_subscriptions.read().unwrap();
-        for (_, sink) in subscriptions.iter() {
-            sink.notify(Ok(SlotInfo { slot, parent, root }))
-                .wait()
-                .unwrap();
+        self.enqueue_notification(NotificationEntry::Slot(SlotInfo { slot, parent, root }));
+    }
+
+    fn enqueue_notification(&self, notification_entry: NotificationEntry) {
+        match self.notification_sender.try_send(notification_entry) {
+            Ok(()) => (),
+            Err(TrySendError::Full(notification)) => {
+                //
+                // NEED FEEDBACK: what to do when full?
```

Comment:
https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB

Seems like eth has a buffer per connection, and closes the connection when they are overloaded as a signal to the client rather than silently dropping notifications. I like that model better as the client is notified that they are no longer getting recent notifications.

Reply:
@carllin that sounds like a neat idea that grows the scope a bit - let me catch up with @mvines to see how he'd want to approach the time investment for design/implementation, I'm currently working on CLI fork visualization and I'm not quite sure if he has other priorities for me to jump on. Thank you again for the suggestion!

Reply:
Yeah nice idea.   Totally a follow-up PR IMO, please file an issue for this!

For this PR I think we should avoid using `sync_channel()` just use normal unbounded `channel()`, as this doesn't seem to be the right place to rate limit regardless.   On receiver side, process_notifications() I think we'd want to pull off all messages at once, sort them into individual queues for each websocket and transmit them independently (hanging up on those who can't keep up) --- in that follow-up PR that is.

Reply:
Issue submitted here: https://github.com/solana-labs/solana/issues/7844